input/lightcurves/desy_lc.py

```diff
@@ -105,49 +119,76 @@ def process_one_file(source_id, filename):
                     second_index = second_index + 2
                 else:
                     break
-    # print('Filenames: ')
-    # print(len(filenames))
-
+    # print(filenames)
 
     # seperate by paper, set metadata
     tables = []
     for x in range(0, len(index) - 1):
         table = tab[index[x]:index[x + 1]]
-        table.meta['source_id'] = str(source_id)
+        table.meta['source_id'] = int(source_id)
         table.meta['telescope'] = ''.join(str(telescopes[x]).split())
-        table.meta['paper_id'] = str(papers[x])
+        table.meta['paper_id'] = ''.join(str(papers[x]).split())
+        if filenames[x][:len(none_paper_id)] == none_paper_id:
+            table.meta['data_id'] = str(none_paper_id)
+        else:
+            table.meta['data_id'] = str('ADS_Bibcode')
         del table['telescope']
         del table['paper']
         tables.append(table)
     table = tab[index[-1] + 1:]
-    table.meta['source_id'] = str(source_id)
+    table.meta['source_id'] = int(source_id)
     table.meta['telescope'] = ''.join(str(telescopes[-1]).split())
-    table.meta['paper_id'] = str(papers[x])
+    table.meta['paper_id'] = ''.join(str(papers[-1]).split())
+    if filenames[-1][:len(none_paper_id)] == (none_paper_id):
+        table.meta['data_id'] = none_paper_id
+    else:
+        table.meta['data_id'] = str('ADS_Bibcode')
     del table['telescope']
     del table['paper']
     tables.append(table)
-    # print(tab)
     # print(tables)
-    # print('Tabellen: ')
-    # print(len(tables))
-    # print('Papers: ')
-    # print(len(papers))
 
-    # write files
-    for x in range(0, len(tables)):
-        tables[x].write(source_id + '_' + filenames[x] + '.ecsv', format='ascii.ecsv')
+    # ceate folder structure and write files
+    count_files = 0
+    paper_repo = Path('/afs/ifh.de/group/amanda/scratch/wegenmat/gamma-cat/input/papers') # will be changed to 'input/papers' if the script is called from top-level repo
+    for x in range(0, len(filenames)):        
+        # create folder structure for none_paper_id
+        if tables[x].meta['data_id'] == none_paper_id:
+            if Path(str(paper_repo / none_paper_id / 'lightcurves' / 'source_id_') + source_id).exists() is False:
```

Comment:
Here you're writing this three times:
```
Path(str(paper_repo / none_paper_id / 'lightcurves' / 'source_id_') + source_id)
```
To avoid code duplication, you can put a local variable and use it three times.
Also, it's possible to create the Path object in a bit simpler way:
```python
path = paper_repo / none_paper_id / 'lightcurves/source_id_{}'.format(source_id)
```